com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/model/DeployedMta.java

```diff
@@ -63,12 +65,17 @@ public boolean equals(Object object) {
         return Objects.equals(metadata, other.metadata);
     }
 
-    public DeployedMtaModule findDeployedModule(String moduleName) {
+    public DeployedMtaApplication findLiveApplication(String moduleName, DeployedMtaApplication.ProductizationState state) {
         return getModules().stream()
-                           .filter(module -> module.getModuleName()
-                                                   .equalsIgnoreCase(moduleName))
+                           .filter(deployedApplication -> doesDeployedApplicationMatch(moduleName, deployedApplication, state))
                            .findFirst()
                            .orElse(null);
     }
 
+    private boolean doesDeployedApplicationMatch(String moduleName, DeployedMtaApplication deployedApplication, ProductizationState state) {
+        return deployedApplication.getModuleName()

```

Comment:
These are two separate checks. Use two filter() calls in the method above.

Reply:
fixed